File: src/main/kotlin/site/siredvin/lib/peripherals/IPeripheralOperation.java
package site.siredvin.lib.peripherals;

import site.siredvin.lib.misc.IConfigHandler;

import java.util.Map;

public interface IPeripheralOperation<T> extends IConfigHandler {
    int getInitialCooldown();
    int getCooldown(T context);
    int getCost(T context);
    Map<String, Object> computerDescription();
}

File: src/main/kotlin/site/siredvin/lib/peripherals/IPeripheralOperation.kt
package site.siredvin.lib.peripherals

import site.siredvin.lib.misc.IConfigHandler

interface IPeripheralOperation<T> : IConfigHandler {
    val initialCooldown: Int
    fun getCooldown(context: T): Int
    fun getCost(context: T): Int
    fun computerDescription(): Map<String, Any?>
}